### ISSUE TITLE
Update is not transparent enough

### ISSUE THREAD

**Issue by karai17** [atom/atom#5554]
When I update Atom, what happens is the Update option turns into an "Updating" message and stays like that indefinitely. I have no way of knowing if the update is actually working, if it is done, or anything else of the sort. It is not until I close and restart Atom that it tells me I am using a new version.

A significant improvement to this would be to do one of the following:

1) Emulate how GitHub for Windows works--silently download the update and prompt the user to install it at their convenience

2) Keep the manual update process but give more information during the process such as % downloaded, % installed, prompt to reopen when done (if necessary), etc.

3) Include Atom updates within the package/theme update system so that you are automatically notified of an update and are provided some GUI information

My vote would probably go with option 1. It's automated and non-intrusive.

**Comment by mark-hahn** [comment]
If the update is being done with squirrel then you are pretty much out of
luck.  Squirrel refuses to do anything like give progress reports.  They
say to just make the load instantaneous (LOL).  I posted an issue
for squirrel.

**Comment by karai17** [comment]
Well then it sounds like squirrel is a pile of expletive and should be thrown out for a competent solution. Giving users information is the most basic thing you can do. Installing Atom from scratch is annoying because it just shows an animated splash image and tells me to trust that it's actually working instead of showing me something simple like a progress bar. The progress bar doesn't even have to be very accurate, something as simple as `FILES_PROCESSED / TOTAL_FILES * 100` would be enough to show that things are moving along and gives you a reasonable estimate of time.

**Comment by mark-hahn** [comment]
>  it just shows an animated splash image and tells me to trust that it's
actually working

That is all squirrel.  I assume it's to late to ditch squirrel and use a
real installer.  I also assume there are some internal features of squirrel
that attracted the core developers.

**Comment by karai17** [comment]
It's never too late, it just may take some serious overhauls which, in my opinion, are necessary. GitHub for Windows has a fantastic update setup and I am very surprised there is this sort of fragmentation within GitHub's own software.

**Comment by 50Wliu** [comment]
This was discussed before in Squirrel/Squirrel.Windows#137 which got closed with none of the suggestions implemented (with the reason being, as @mark-hahn noted, that adding said suggestions would "inform the user just how much we suck" and the proper way to go forward was to make Atom install faster).

Also, I believe Atom is doing 1) right now.  It silently downloads the update in the background and automatically applies it when all Atom processes get closed (or when the Restart and Update button is clicked).

**Comment by paulcbetts** [comment]
Whoa whoa whoa, I think we're conflating two things - most of this thread is talking about the *initial install*, but the initial question was about *updates*. 

Squirrel on Windows *does* provide progress for updates (in the form of percentages written to stdout), but Atom doesn't do anything with this value currently. It certainly could though. 

> Emulate how GitHub for Windows works--silently download the update and prompt the user to install it at their convenience

This isn't how GitHub for Windows works, the update will be installed regardless of whether you choose to install it the next time the app starts. In reality, the only thing that "Install update" button in GH4W is restart the app. 

>  GitHub for Windows has a fantastic update setup and I am very surprised there is this sort of fragmentation within GitHub's own software.

Spoiler alert: I wrote *both* Atom for Windows's installer, and the logic / UI interaction in GitHub for Windows.

**Comment by karai17** [comment]
There should be no particular difference between initial install and updates, I should be given progress data for both.

If Squirrel supports this data and Atom does not utilize it, then my issue still stands in that Atom needs to be more transparent.

I feel like silent updates are great. GitHub for Windows and Chrome both silently update and give you prompt to finalize the install, which is great for keeping up to date without forcing the user to stop what they are doing (cough, Windows Update, cough). I am totally okay with that functionality as it only makes the process faster for the user.

I am even more confused now that I know the same person wrote both systems that seem to act quite differently.

**Comment by mehcode** [comment]
> This isn't how GitHub for Windows works, the update will be installed regardless of whether you choose to install it the next time the app starts. In reality, the only thing that "Install update" button in GH4W is restart the app.

Why aren't we doing this sort of auto update-in-the-background workflow for atom? Just a "will be done later" thing?

**Comment by paulcbetts** [comment]
> Why aren't we doing this sort of auto update-in-the-background workflow for atom? Just a "will be done later" thing?

This is exactly what Atom does on both Mac and Windows

**Comment by 50Wliu** [comment]
Atom does have background updates just like Github for Windows.
Fake edit: Beaten by Paul ;P.

**Comment by karai17** [comment]
I have a hard time believing that. Atom never seems to update unless I manually trigger it, and when I do so it never seems to do anything except consume a whole bunch of CPU. It just sits there and says "Updating" indefinitely without any sort of notification that it is done and needs to be restarted.

**Comment by mark-hahn** [comment]
>  It just sits there and says "Updating"

I started using the command line a long time ago because of this kind of
annoyance.  It is a much better experience.

**Comment by Zireael07** [comment]
Regarding install, I would like to see the progress data shown in some way.

Regarding updates, clicking update tells me 0.182 is the latest one, which is not true as of an hour ago.

**Comment by mark-hahn** [comment]
> clicking update tells me 0.182 is the latest one

The delay can be up to a day or two.

**Comment by rogeriotaques** [comment]
+1

**Comment by mnquintana** [comment]
Closing in favor of https://github.com/atom/atom/issues/9164
